resources/ts/Components/UitIdWidget.tsx

```diff
@@ -88,7 +92,7 @@ export const UitIdWidget = ({
             ></div>
             <a
               className={classNames(i18n.language === "en" && "active")}
-              href={`${translateRoute(`/${currentPage}`, "en")}?setLocale=true`}
+              href={`${translateRoute(`/${currentPage}`, "en", id)}?setLocale=true`}
```

Comment:
Nice solution ⭐ We better make it a bit more generic, since we could have other urls with dynamic parts of the url.
The id passed here can be an object with variables, depending on the page. Then we have to create a mapping between the page (e.g. /detail) and the needed variables (id in this case). Something like:

```ts
const currentPageToVariables = {
    'integrations/detail': props.integration.id
    // other pages come here
}
```